whitelist/src/node/batch_manager/mod.rs

```
@@ -625,6 +508,48 @@ impl BatchManager {
         Ok((l2_block, l2_slot_info))
     }
 
+    async fn add_new_l2_block_to_batch(
+        &mut self,
+        l2_block: L2Block,
+        l2_slot_info: L2SlotInfo,
+        end_of_sequencing: bool,
+        operation_type: OperationType,
+        // When we previously add a forced inclusion block, we don't want to return an error
+        // because we don't want to break verify_preconfed_block
+        can_return_error: bool,
+    ) -> Result<Option<BuildPreconfBlockResponse>, Error> {
+        let anchor_block_id = self
+            .batch_builder
+            .add_l2_block_and_get_current_anchor_block_id(l2_block.clone())?;
+
+        match self
+            .taiko
+            .advance_head_to_new_l2_block(
+                l2_block,
+                anchor_block_id,
+                &l2_slot_info,
+                end_of_sequencing,
+                false,
+                operation_type,
+            )
+            .await
+        {
+            Ok(preconfed_block) => Ok(preconfed_block),
+            Err(err) => {
+                error!("Failed to advance head to new L2 block: {}", err);
+                self.remove_last_l2_block();
+                if can_return_error {
```

I think it's not a good way of handling it. Instead, it should return an error, but the function which call it the error should decide what to do with it - ignore or stop some further execution.

---

Agree